cpp/include/cudf/strings/string_view.cuh

```diff
@@ -373,24 +373,23 @@ __device__ inline size_type string_view::find_impl(char const* str,
                                                    size_type pos,
                                                    size_type count) const
 {
-  auto const nchars = length();
-  if (!str || pos < 0 || pos > nchars) return npos;
-  if (count < 0) count = nchars;
+  if (!str || pos < 0) { return npos; }
```

Comment:
checking for `null` or `pos < 0` might be redundant, as I don't think it's worth it checking for invariant violation.
the `bytes` should also already specify the length of the `str`, even if it's signed.
if you feel it should exist then we should add an `unlikely` attribute.
`string_view::compare` for example doesn't check for nulls or that the size invariant is violated.

Reply:
These are definitely unlikely but not expensive. How do I add an `unlikely` attribute?